### ISSUE TITLE
A few issues with syntax highlighting

### ISSUE THREAD

**Issue by danielbmarques** [lervag/vimtex#491]
There are a few details that I think could be improved in Vimtex's syntax highlighting features:

1. It seems to disable spelling for words inside `\emph` commands. Here's the same file with and without Vimtex:

    ![captura de tela 2016-07-13 as 14 48 29](https://cloud.githubusercontent.com/assets/15813674/16813810/9209ee04-4909-11e6-9a17-09a444a627b4.png) ![captura de tela 2016-07-13 as 14 48 58](https://cloud.githubusercontent.com/assets/15813674/16813812/947fb128-4909-11e6-90a8-7c1d51b91b8f.png)

2. This could be my fault, but Vimtex overrides what I set in `~/.vim/after/syntax/tex.vim`?
    I have these lines in that file:
    ```vim
    syntax match texStatement "\\citetitle" nextgroup=texRefOption,texCite
    syntax match texStatement "\\apud" nextgroup=texRefOption,texCite
    ```
   The first is for a standard `biblatex` command and the second is for a command used in my own [`biblatex-abnt`](https://github.com/abntex/biblatex-abnt) style.

    Without Vimtex it works perfectly:
   ![captura de tela 2016-07-13 as 14 49 09](https://cloud.githubusercontent.com/assets/15813674/16814781/b3df614a-490d-11e6-92c5-6a30ac210ebd.png)

    With Vimtex it doesn't:
    ![captura de tela 2016-07-13 as 14 47 19](https://cloud.githubusercontent.com/assets/15813674/16814795/c1f94df4-490d-11e6-992d-36890d134348.png)

    Should Vimtex override the stuff in `~/.vim/after/syntax/tex.vim`? If so, where should I add those lines?

3. And this brings me to issue number three: most of the standard `biblatex` commands aren't supported. Here's a full list of the commands mentioned in the manual:

    ![captura de tela 2016-07-13 as 15 27 31](https://cloud.githubusercontent.com/assets/15813674/16814926/598c0d64-490e-11e6-97c7-0040dd79ef25.png)

    Do you think I should talk to ~~Bernd Feige~~ Charles Campbell about adding this to the default syntax file? I ask because you have your own "Improve support for cite commands" part at `/vimtex/after/syntax/`, so there might be a reason to keep this here. I could try to add this stuff myself and make a PR, but I'm not big on regex, so I'd probably have to use many keywords to account for commands that could be abstracted into a single pattern. Let me know if you want me to try anyway.

**Comment by lervag** [comment]
Hi @danielbmarques Thanks, I've fixed 1. immediately, as that was trivial. As for your other issues:

2. This is because of the runtime order. That is, for some reason, your after script is loaded first. If I create a file `~/.vim/after/syntax/tex.vim`, add `echom 'new'`, then add `echom 'vimtex'` in the corresponding file in vimtex, then the order of the echo statements show that in my case, `vimtex` is correctly loaded first. I assume this is due to the plugin manager. I use [vim-plug](https://github.com/junegunn/vim-plug).

3. I think taking to Charles Campbell would be the best solution, because if he adds support, then it will benefit all Vim users. Thus, I strongly recommend that you contact him. In the meantime, I can add support for this in vimtex which I can remove if this ever comes to the default syntax file. I can do the work myself, but it would help if you could provide a list of the commands that I can copy/paste. Also, if they work differently from other cite commands, it would also help to get examples of how the commands are used.

**Comment by danielbmarques** [comment]
Thanks for the answer. `vim-plug` looks great, I'll give it a try.

This is the file I used in my example:

[biblatex_commands.txt](https://github.com/lervag/vimtex/files/364772/biblatex_commands.txt)
(Except the actual extension was `tex`, but github doesn't allow attaching that.)

I'm not sure if I didn't make any typos or skipped any commands, I did that in a hurry, but I'll check again when I have the time. Also I don't know how many of them work. The ones ending with an "s" can take a lot more arguments, in case that's relevant:

    \cite[⟨prenote⟩][⟨postnote⟩]{⟨key⟩}

    \cites(⟨multiprenote⟩)(⟨multipostnote⟩)[⟨prenote⟩][⟨postnote⟩]{⟨key⟩}...[⟨prenote⟩][⟨postnote⟩]{⟨key⟩}

The commands until the `\citeurl` command in that file (reading from left to right regardless of the columns) work mostly like that I think. After the `\citeurl` command there are some weird ones that I don't know well, we'd have to check [the manual](http://tug.ctan.org/macros/latex/exptl/biblatex/doc/biblatex.pdf) (section 3.8.6).

Since you're going to add it here I'll wait until we have something and I'll send it to Charles Campbell then, ok?

**Comment by danielbmarques** [comment]
I just found a new weird issue:

![vimtex](https://cloud.githubusercontent.com/assets/15813674/16866811/8a0e4d9e-4a44-11e6-8293-7b2092974b75.gif)

This is a tex file with just `vimtex` and `neocomplete` enabled. Without `vimtex`, as soon as I start typing the highlighting for the parentheses disappears. With `vimtex` the highlighting remains, and then there's that weird problem, one of the letters is highlighted instead of the closing parenthesis.

I don't think this was happening a few days ago, but I'm not 100% sure. Also I just switched to `vim-plug` and apparently my files weren't being sourced in the correct order before, I don't know if that could have had something to do with it.

**Comment by danielbmarques** [comment]
I just found a new weird issue:

![vimtex](https://cloud.githubusercontent.com/assets/15813674/16866811/8a0e4d9e-4a44-11e6-8293-7b2092974b75.gif)

This is a tex file with just `vimtex` and `neocomplete` enabled. Without `vimtex`, as soon as I start typing the highlighting for the parentheses disappears. With `vimtex` the highlighting remains, and then there's that weird problem, one of the letters is highlighted instead of the closing parenthesis.

I don't think this was happening a few days ago, but I'm not 100% sure. Also I just switched to `vim-plug` and apparently my files weren't being sourced in the correct order before, I don't know if that could have had something to do with it. Also it only happens with `syntax on` (the parentheses are highlighted even without it), so I think it must have something to do with the syntax files.

**Comment by lervag** [comment]
That's strange. This works as expected on my end. I would be happy if you could open a new issue with a minimal description of how to reproduce this. That will allow us to use this issue to continue the discussion about biblatex highlighting.

**Comment by lervag** [comment]
I looked in the manual, so I think the latest version should provide relatively complete support for biblatex commands. Let me know what you think.

Feel free to send the suggestion to Charles Campbell, and don't hesitate to send the code. It might not be perfect, but it seems to work well for me. I've attached the file I used to test with (and similar to you, I've used `txt` extension instead of `tex` extension).

[test.txt](https://github.com/lervag/vimtex/files/366672/test.txt)
